src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -1015,14 +1015,17 @@ private async Task HandshakeAsync(ConnectionState startingConnectionState, Cance
             if (sendHandshakeResult.IsCompleted)
             {
                 // The other side disconnected
-                throw new InvalidOperationException("The server disconnected before the handshake was completed");
+                var ex = new IOException("The server disconnected before the handshake could be started.");
+                Log.ErrorReceivingHandshakeResponse(_logger, ex);
+                throw ex;
             }
 
             var input = startingConnectionState.Connection.Transport.Input;
 
+            using var handshakeCts = new CancellationTokenSource(HandshakeTimeout);
```

Comment:
Curious, why did you move this?

Reply:
I log different messages when the user manually cancels the handshake using a CancellationToken or by calling StopAsync() than I do when the handshake simply times out. I moved this out of the try scope so I could check `handshakeCts.IsCancellationRequested` in the catch block below to determine what to log.